pkg/driver/controllerserver.go

```diff
@@ -153,7 +167,10 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	}
 
 	var deleteErr error
-	if prefix == "" {
+	if usePrefix, err := strconv.ParseBool(client.GetFSMetaField(meta, mounter.UsePrefix)); err == nil && usePrefix {
```

Comment:
I don't quite understand what `GetFSMetaField` is for, you can simply access the values from `meta`, no? Especially if we would use a bool for `meta.UsePrefix` (see other comment about that).